Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -1895,15 +1895,33 @@ bool WebsiteDataStore::isBlobRegistryPartitioningEnabled() const
     });
 }
 
-void WebsiteDataStore::updateBlobRegistryPartitioningState()
+bool WebsiteDataStore::isOptInCookiePartitioningEnabled() const
+{
+    return WTF::anyOf(m_processes, [] (const WebProcessProxy& process) {
```

Comment:
Why do we want a per-page setting if one page could affect all the other pages in the same session?

Reply:
This should only be used for testing, and applications like Safari only expose a single setting for all pages. The idea is that when the preference is enabled on any page, then we enable it in the NetworkSession. We only disable it in the NetworkSession when it is disabled on all pages.